### ISSUE TITLE
Package does not pass AreTheTypesWrong

### ISSUE THREAD

**Issue by jakebailey** [ArnaudBarre/eslint-plugin-react-refresh#70]
See: https://arethetypeswrong.github.io/?p=eslint-plugin-react-refresh%400.4.16

The package is bundled, and ends with:

```js
var src_default = { rules, configs };
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  configs,
  rules
});
```

This means that the output dts file should actually read `export = _default;`, not `export default _default;`, since an export named `default` does not actually exist at runtime.

I noticed this on DT, which previously held types for this package (but will have to be deleted now that you're doing it): https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/12118993180/job/33784693075#step:12:2663

**Comment by jakebailey** [comment]
Er, sorry, misread esbuild's output.

```
var src_exports = {};
__export(src_exports, {
  configs: () => configs,
  default: () => src_default,
  rules: () => rules
});
module.exports = __toCommonJS(src_exports);
```

I _think_ it actually needs `export = _default` and `export { _default as default }`?

**Comment by jakebailey** [comment]
I should really just defer to the ATTW message 😅 

> The JavaScript appears to set both module.exports and module.exports.default for improved compatibility, but the types only reflect the latter (by using export default). This will cause TypeScript under the node16 module mode to think an extra .default property access is required, which will work at runtime but is not necessary. These types export = an object with a default property instead of using export default.

**Comment by ArnaudBarre** [comment]
Oh I didn't know someone published types for this package.
I'll update the suggested syntax

**Comment by jakebailey** [comment]
My suggestion is probably wrong; you may need to declare a different type instead that's an object type with a default prop and not mix syntax. I think TS will error?

**Comment by ArnaudBarre** [comment]
Finally took some time to do the change suggested by ATTW cli, it was green after the change.

**Comment by jakebailey** [comment]
I'm not sure that change is exactly correct, though it does pass ATTW; the runtime code impies that code like:

```js
const plugin = require("eslint-plugin-react-refresh");

plugin.configs;
plugin.rules;
```

Will work, but the new types imply that the exported object _only_ has a prop named `default`.

![image](https://github.com/user-attachments/assets/78c25f00-dfbb-4a4b-b845-be237f7a55d0)

**Comment by jakebailey** [comment]
(I had forgotten about this; when I have some time later I can try and come up with a "real" declaration for the shape of your runtime export.)

**Comment by ArnaudBarre** [comment]
export = _default both is valid for the cli and provide correct typing for moduleResolution bundler and node16 (with `allowSyntheticDefaultImports`).

I think this is good enough for me!

**Comment by jakebailey** [comment]
I'm testing with node16 in a plain CJS file, where those synthetic defaults and such don't work since TS wouldn't transform them (which should be the case for nearly all eslint configs).
